### ISSUE TITLE
Scrolling down and clicking on an item selects the wrong item

### ISSUE THREAD

**Issue by tomusiaka** [Pixabay/jQuery-autoComplete#24]
I am using Chrome on MacOS and when I click on the field to see suggestions, scroll down to some suggestion and click on it, it selects the suggestion that corresponds to the one that would have been clicked if I never scrolled. Simplified example: if there were items 1, 2, 3 and 4 and I can see only 1 and 2, and scrolling all the way down shows 3 and 4, then clicking on 3 will select 1 and clicking on 4 will select 2. Scrolling mid-way so that 2 and 3 are visible would cause clicking on 2 to select 1 and clicking on 3 to select 2. Let me know if it makes sense.

I can reproduce this issue on your demo page. It appears that on mouse down it scrolls all the way up, which is hard to notice on click.

This issue can be reproduces about 50% of the time.

**Comment by SimonSteinberger** [comment]
Not having a Mac around, this is hard to debug for me right now. If you find the cause yourself, please post it, so I can include the solution in the repo. I'll try to check this out when having access to a Mac.

**Comment by tomusiaka** [comment]
I was able to figure this out. When being focused on the text field on mouse down on a suggested item, `blur.autocomplete` is triggered. I believe it is triggered automatically somehow by jQuery since I don't see any trigger in your code. Then, this line:

https://github.com/Pixabay/jQuery-autoComplete/blob/master/jquery.auto-complete.js#L87

tries to focus back on the text field. Focusing in turn executes the code that refreshes the suggestions and scrolls to the top. Then, mouse up selects the wrong suggestion because it's scrolled to the top.

Removing that line 87 and adding `that.focus();` after line 79 fixed the problem for me.

**Comment by SimonSteinberger** [comment]
Just tried. Unfortunately, in various browsers, using scroll bars in the suggestions box removes focus from the input field - and it's not returned by `that.focus();` after line 79. It seems to be a race/timing problem. So, depending on the browser/system, the bug is there or not. `blur`is triggered by the browser here.

You could try using `mousedown` instead of `mouseup` in the original code. Maybe that solves the race issue. Not seeing the issue myself, I cannot test it here.

**Comment by ecoreng** [comment]
adding:
```
that.sc.on("mousedown", function(ev){
  return false;
});
```

just before
```
that.on('blur.autocomplete', function(){
```

solves the issue for me on Chrome-Ubuntu and Win7-IE10/IE9

it might even solve this issue:
https://github.com/Pixabay/jQuery-autoComplete/issues/40

**Comment by SimonSteinberger** [comment]
Interesting suggestion, will try :-1:

**Comment by SimonSteinberger** [comment]
Ok, I've included this patch. The focus handler was already bound, but I added `return false;` to the end of it. Hope that works, because I couldn't actually reproduce this bug. But I also didn't notice any disadvantage of the patch.

It would be interesting of course, if that indeed also fixes #40.

**Comment by ecoreng** [comment]
If this doesn't fix the other issue, I would try to bind a touchstart event <or similar> with a ``return false;`` too. I think the problem is of the same nature but I don't have an IE touch device around to test.
